src/action/network.rs

```diff
@@ -0,0 +1,297 @@
+        TcpState::Established => State::Established,
+        TcpState::FinWait1 => State::FinWait1,
+        TcpState::FinWait2 => State::FinWait2,
+        TcpState::CloseWait => State::CloseWait,
+        TcpState::Closing => State::Closing,
+        TcpState::LastAck => State::LastAck,
+        TcpState::TimeWait => State::TimeWait,
+        TcpState::DeleteTcb => State::DeleteTcb,
+    }
+}
+
+/// Creates `NetworkConnection` protobuf message from
+/// [netstat2::ProtocolSocketInfo][protocol_socket_info]. The fields that are
+/// impossible to fill using the socket information are set to `None`.
+///
+/// [protocol_socket_info]: ../../../netstat2/enum.ProtocolSocketInfo.html
+fn make_connection_from_socket_info(
+    socket_info: &ProtocolSocketInfo
+) -> NetworkConnection {
+    use ProtocolSocketInfo::{Tcp, Udp};
+
+    match socket_info {
+        Tcp(tcp_info) => NetworkConnection {
+            family: Some(make_family(&tcp_info.local_addr).into()),
+            r#type: Some(Type::SockStream.into()),
+            local_address: Some(make_network_endpoint(
+                &tcp_info.local_addr,
+                tcp_info.local_port
+            )),
+            remote_address: Some(make_network_endpoint(
+                &tcp_info.remote_addr,
+                tcp_info.remote_port
+            )),
+            state: Some(make_state(&tcp_info.state).into()),
+            ..Default::default()
+        },
+        Udp(udp_info) => NetworkConnection {
+            family: Some(make_family(&udp_info.local_addr).into()),
+            r#type: Some(Type::SockDgram.into()),
+            local_address: Some(make_network_endpoint(
+                &udp_info.local_addr,
+                udp_info.local_port
+            )),
+            remote_address: None,
+            state: None,
+            ..Default::default()
+        },
+    }
+}
+
+impl super::Request for Request {
+
+    type Proto = ListNetworkConnectionsArgs;
+
+    fn from_proto(proto: Self::Proto) -> Result<Request, session::ParseError> {
+        Ok(Request {
+            listening_only: proto.listening_only.unwrap_or(false),
+        })
+    }
+}
+
+impl super::Response for Response {
+
+    const RDF_NAME: Option<&'static str> = Some("NetworkConnection");
+
+    type Proto = NetworkConnection;
+
+    fn into_proto(self) -> Self::Proto {
+        let mut result: NetworkConnection;
+        result = make_connection_from_socket_info(&self.socket_info);
+        if let Some(process_info) = self.process_info {
+            result.pid = Some(process_info.pid);
+            result.process_name = process_info.name;
+        }
+        result
+    }
+}
+
+impl From<&Process> for ProcessInfo {
+
+    fn from(process: &Process) -> ProcessInfo {
+        ProcessInfo {
+            pid: process.pid() as u32,
+            name: Some(process.name().to_string()),
+        }
+    }
+}
+
+impl ProcessInfo {
+
+    /// Constucts a ProcessInfo system from a given process ID. The process
```

Comment:
Missing backticks to denote `ProcessInfo` as an identifier.

Also, I would move the next sentence to next paragraph. I tend to reserve first paragraph for short, single-sentence summary (because it is what is visible on the list of functions in the generated documentation), while leaving the details for the following paragraphs.